pystan/stanfit4model.pyx

```diff
@@ -401,15 +401,24 @@ def _call_sampler(data, args, pars_oi=None):
         pars_oi_bytes = [n.encode('ascii') for n in pars_oi]
         if len(pars_oi_bytes) != fitptr.param_names_oi().size():
             fitptr.update_param_oi(pars_oi_bytes)
-    ret = fitptr.call_sampler(deref(argsptr), deref(holderptr))
+
+    # Print any runtime exception but don't crash out
+    call_sampler_error = None
+    ret = 0
+    try:
+        ret = fitptr.call_sampler(deref(argsptr), deref(holderptr))
+    except RuntimeError as e:
+        print('{}: {}'.format(type(e).__name__, str(e)))
+        call_sampler_error = e
+
     holder = _pystanholder_from_stanholder(holderptr)
     # FIXME: rather than fetching the args from the holderptr, we just use
     # the argsptr we passed directly. This is a hack to solve a problem
     # that holder.args gets dropped somewhere in C++.
     holder.args = _dict_from_stanargs(argsptr)
     del argsptr
     del fitptr
-    return ret, holder
+    return ret, holder, call_sampler_error
```

Comment:
Other function using this function also need update. Maybe currently reraise the error (`sampling`, `vb`). There could be output to logger in python side before raising the error?

Reply:
Sorry, I'm not sure I understand the problem. I can stop passing the actual error object to the python code to preserve compatibility, and use just the return code instead. Is that the fix you are talking about?

Reply:
this now returns 3 parameters, so also sampling and vb are getting these, so I think you can add extra keyword to grab the error and raise it if error was raised.

So 

optimize --> error --> warn in python --> return results (optional)
sampling --> error --> raise error in python
vb --> error --> raise error in python

(I probably need to double check what I'm trying to say)